File: exercism.io/kotlin/space-age/src/main/kotlin/SpaceAge.kt
class SpaceAge(s: Long) {
    val seconds = s
    val earthYears = s / 31557600.0

    fun onEarth() : Double = convert(earthYears)
    fun onMercury() : Double = convert(earthYears / 0.2408467)
    fun onVenus() : Double = convert(earthYears / 0.61519726)
    fun onMars() : Double = convert(earthYears / 1.8808158)
    fun onJupiter() : Double = convert(earthYears / 11.862615)
    fun onSaturn() : Double = convert(earthYears / 29.447498)
    fun onUranus() : Double = convert(earthYears / 84.016846)
    fun onNeptune() : Double = convert(earthYears / 164.79132)

    fun convert(d: Double): Double = Math.round(d * 100) / 100.0
}

File: exercism.io/cpp/space-age/space_age.cpp
#include "space_age.h"

namespace space_age {

long long space_age::seconds() const {
    return earth_seconds;
}

double space_age::on_mercury() const {
    return earth_years / 0.2408467;
}

double space_age::on_venus() const {
    return earth_years / 0.61519726;
}

double space_age::on_earth() const {
    return earth_years;
}

double space_age::on_mars() const {
    return earth_years / 1.8808158;
}

double space_age::on_jupiter() const {
    return earth_years / 11.862615;
}

double space_age::on_saturn() const {
    return earth_years / 29.447498;
}

double space_age::on_uranus() const {
    return earth_years / 84.016846;
}

double space_age::on_neptune() const {
    return earth_years / 164.79132;
}

}  // namespace space_age
